src/core/include/openvino/core/op_extension.hpp

```diff
@@ -33,12 +33,39 @@ class OPENVINO_API BaseOpExtension : public Extension {
      */
     virtual ov::OutputVector create(const ov::OutputVector& inputs, ov::AttributeVisitor& visitor) const = 0;
 
+    /**
+     * @brief Returns extensions that should be registered together with this extension class object.
+     *
+     * Attached extensions may include frontend extensions that OpenVINO op to framework ops or necessary
+     * transformations that should be applied to the network which consist of target op.
+     *
+     * @return
+     */
+    virtual std::vector<ov::Extension::Ptr> get_attached_extensions() const {
+        return {};
+    }
+
     /**
      * @brief Destructor
      */
     ~BaseOpExtension() override;
 };
 
+namespace detail {
+
+#define OV_COLLECT_ATTACHED_EXTENSIONS(FRAMEWORK)                                                         \
```

Comment:
Why do we have framework specific in this header?

Reply:
discussed offline
I will try to move fw mention to frontend part

Reply:
As a result of discussion we decided investigate this as part of separate ticket.
ticket: 77980